x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InternalInferenceAggregation.java

```diff
@@ -48,24 +51,44 @@ public InternalAggregation reduce(List<InternalAggregation> aggregations, Reduce
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public Object getProperty(List<String> path) {
-        Map<String, Object> resultMap = this.inferenceResult.writeResultToMap();
-
-        for (int i=0; i<path.size() -1; i++) {
-            Object value = resultMap.get(path.get(i));
-            if (value == null) {
-                throw new InvalidAggregationPathException("Cannot find an key [" + path.get(i) + "] in " + path);
-            }
-
-            if (value instanceof Map<?, ?>) {
-                resultMap = (Map<String, Object>)value;
+        if (path.isEmpty()) {
+            return this;
+        } else if (path.size() == 1) {
+            String field = path.get(0);
+            if (CommonFields.VALUE.getPreferredName().equals(field)) {
+                if (inferenceResult instanceof ClassificationInferenceResults) {
+                    return ((ClassificationInferenceResults)inferenceResult).transformedPredictedValue();
+                } else if (inferenceResult instanceof SingleValueInferenceResults) {
+                    return ((SingleValueInferenceResults)inferenceResult).value();
+                } else {
+                    return null;
+                }
+            } else if (SingleValueInferenceResults.FEATURE_IMPORTANCE.equals(field)) {
+                if (inferenceResult instanceof SingleValueInferenceResults) {
+                    SingleValueInferenceResults valueResult = (SingleValueInferenceResults) inferenceResult;
+                    return valueResult.getFeatureImportance();
+                } else {
+                    return null;
+                }
```

Comment:
awe snap! Since we are handling the paths...I wonder if we could handle paths like `feature_importance.0.importance` and `feature_importance.0.feature_name`

Reply:
The original implementation I had converted the result to a map which the code would navigate through using the split path so it was more generic as you suggested. I actually changed that so I could take a few lines of code out of this PR.

The only use of this method I could find is in a [Bucket Selector](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-pipeline-bucket-selector-aggregation.html) which will only work with Doubles so I though this was sufficient in the first case. 

I'm open to the change though, maybe in the follow up PR?

Reply:
Follow up PR would be good. 

Bucket select for when the most important feature is > X or when top class probability/score is > X would be neat!